godotproject/Player.gd

```diff
@@ -406,9 +407,26 @@ func _physics_process(delta):
 			velocity_after_swing += velocity.normalized() * tongue_exit_jump_bonus_speed_in_dirn
 			velocity_after_swing += Vector2.UP * tongue_exit_jump_bonus_speed_up
 			velocity = velocity_after_swing
-		elif tongue_pressed:
-			# When we exit swinging using the tongue button, just drop them out of swinging
+		
+		if tongue_pressed:
+			# When we exit swinging using the tongue button, shoot player in direction of tongue
+			# with small upwards boost
 			stop_swing()
+			
+			# A unit vector in the direction of the mouth to the tip of tongue.
+			# It will be used to calculate the launch direction.
+			# The direction is calculated as the vector perpendicular to the current velocity,
+			# we divide by the magnitude to make it a unit vecotr. This helps to normalize it.
+			var tongue_direction = Vector2(velocity.y, -1*velocity.x).normalized()
+			
+			# Make sure this always launches the player in the upwards direction
+			if tongue_direction.y > 0:
+				tongue_direction = tongue_direction * -1
+			
+			# Set velocity
+			var velocity_after_swing = tongue_direction * (200 + 1 	* swing_radius) #TODO make these variables
+			velocity_after_swing += Vector2.UP * tongue_exit_launch_bonus_speed_up # gives it a little oomf
```

Comment:
you toggle the factors here and lmk what you think feels smooth

Reply:
I played around a bit with the values that are already here, and they felt pretty good. Just don't forget about the TODO!